src/data_provider/src/packages/packagesPYPI.hpp

```diff
@@ -0,0 +1,154 @@
+            packageInfo["format"] = "pypi";
+            packageInfo["source"] = UNKNOWN_VALUE;
+            packageInfo["location"] = path.string();
+            packageInfo["priority"] = UNKNOWN_VALUE;
+            packageInfo["size"] = 0;
+            packageInfo["vendor"] = UNKNOWN_VALUE;
+            packageInfo["install_time"] = UNKNOWN_VALUE;
+            packageInfo["multiarch"] = UNKNOWN_VALUE;
+
+            TFileIO::readLineByLine(path,
+                                    [&packageInfo](const std::string & line) -> bool
+            {
+                const auto it {
+                    std::find_if(PYPI_FIELDS.begin(),
+                                 PYPI_FIELDS.end(),
+                                 [&line](const auto & element)
+                    {
+                        return line.find(element.first) != std::string::npos;
+                    })};
+
+                if (PYPI_FIELDS.end() != it)
+                {
+                    const auto& [key, value] {*it};
+                    packageInfo[value] = line.substr(key.length());
+                }
+                return true;
+            });
+
+            // Check if we have a name and version
+            if (packageInfo.contains("name") && packageInfo.contains("version"))
+            {
+                callback(packageInfo);
+            }
+        }
+
+        void findCorrectPath(const std::filesystem::path& path, std::function<void(nlohmann::json&)> callback)
+        {
+            try
+            {
+                const auto filename = path.filename().string();
+
+                for (const auto& [key, value] : FILE_MAPPING_PYPI)
+                {
+                    if (filename.find(key) != std::string::npos)
+                    {
+                        if (TFileSystem::is_regular_file(path))
+                        {
+                            parseMetadata(path, callback);
+                        }
+                        else if (TFileSystem::is_directory(path))
+                        {
+                            parseMetadata(path / value, callback);
+                        }
+                        else
+                        {
+                            // Do nothing
+                        }
+                    }
+                }
+            }
+            catch (const std::exception& e)
+            {
+                std::cerr << "Error parsing PYPI package: " << path.string() << ", " << e.what() << std::endl;
+            }
+        }
+        void exploreExpandedPaths(const std::vector<std::string>& expandedPaths,
+                                  std::function<void(nlohmann::json&)> callback)
+        {
+            for (const auto& expandedPath : expandedPaths)
+            {
+                try
+                {
+                    // Exist and is a directory
+                    if (TFileSystem::exists(expandedPath) && TFileSystem::is_directory(expandedPath))
```

Comment:
If Utils::expandAbsolutePath supports the type of file, then this conditional would be unnecessary, and also the loop cycles will be reduced.

Reply:
the idea is only to identify folders in the pypi implementation here.